core/restore.php

```diff
@@ -0,0 +1,418 @@
+<?php
+/**
+ * @file
+ * Administrative page for restoring backup database and configuration files.
+ *
+ * Point your browser to "http://www.example.com/core/restore.php" and follow
+ * the instructions.
+ *
+ * If you are not logged in using either the site maintenance account or an
+ * account with the "Restore site backups" permission, you will need to
+ * modify the access check statement inside your settings.php file. After
+ * finishing the upgrade, be sure to open settings.php again, and change it
+ * back to its original state!
+ */
+
+/**
+ * Defines the root directory of the Backdrop installation.
+ */
+define('BACKDROP_ROOT', dirname(dirname($_SERVER['SCRIPT_FILENAME'])));
+
+// Change the directory to the Backdrop root.
+chdir(BACKDROP_ROOT);
+
+/**
+ * Global flag indicating that restore.php is being run.
+ *
+ * When this flag is set, various operations do not take place, such as invoking
+ * hook_init() and hook_exit(), css/js preprocessing, and translation.
+ */
+const MAINTENANCE_MODE = 'restore';
+
+/**
+ * Provides an overview of the backup restoration process.
+ *
+ * This page provides cautionary statements before proceeding with a restore.
+ *
+ * @return string
+ *   Rendered HTML form.
+ */
+function restore_info_page() {
+  restore_task_list('info');
```

Comment:
I wonder, if it might make sense to prevent this page entirely.

Currently it's available (based on permissions) even if `$settings['backup_directory']` is either an empty string or FALSE. It just leads to an empty "Select backup" listing.

For example if the settings FALSE, return a 403 instead? Or would this be too confusing and the empty listing provides better UX?

Reply:
Ah, thanks I meant to make a better page for that! What it should do is show you the page (if you have access or `restore_free_access = TRUE`) and instead inform you of the purpose and link to documentation, rather than just showing an empty list.

Reply:
Hm. The restore_free_access info works just fine. What doesn't seem to be considered is, that the backup feature is off entirely. The page is still accessible. That does no harm, but doesn't provide any value, either.

TBH, I'm not 100% sure, what I'd expect to find on `core/restore.php` (depending on permissions), if backups are completely off.

Reply:
`update.php` provides somewhat useful information on its 403 page (though if you had admin access it might not be a 403 for restore.php):

![image](https://github.com/user-attachments/assets/10ab099b-e4f8-4b5e-8e6b-db6c4026ff92)

Reply:
I made an alternative page if `$settings['backup_directory']` is not specified:

![image](https://github.com/user-attachments/assets/937a6a7d-418a-4925-916f-df5e6db81b31)